src/parser/core/modules/Defensives.tsx

```diff
@@ -35,6 +35,11 @@ export class Defensives extends Analyser {
 	 * Implementing modules should provide a list of job-specific defensive actions to track
 	 */
 	protected trackedDefensives: Action[] = []
+
+	// Private lists of the cooldown usage and charge histories so we don't keep recalculating it via calls to the cooldowns module
+	private cooldownHistories: {[key: number]: CooldownHistoryEntry[]} = {}
+	private chargeHistories: {[key: number]: ChargeHistoryEntry[]} = {}
```

Comment:
caching these is technically fine, because this module doesn't generate anything until the rest of the parse is complete - but worth documenting as such.

if performance is a concern, i'd be inclined to move caching down to the source (i.e. the cooldowns module), as it allows cache _breaking_ to be colocated.

Reply:
Since we're doing some filtering on the endReason prop, I think we can leave this here for now but it might not hurt to update the cooldowns module to give those options in the future.